src/AutoRemoteSyscalls.cc

```diff
@@ -326,135 +326,151 @@ static void write_socketcall_args(Task* t, remote_ptr<void> remote_mem,
   t->write_mem(remote_mem.cast<socketcall_args<Arch>>(), sc_args, ok);
 }
 
-static size_t align_size(size_t size) {
-  static int align_amount = sizeof(uintptr_t);
-  return (size + align_amount - 1) & ~(align_amount - 1);
-}
-
-static remote_ptr<void> allocate(remote_ptr<void>* buf_end,
-                                 const AutoRestoreMem& remote_buf,
-                                 size_t size) {
-  remote_ptr<void> r = *buf_end;
-  *buf_end += align_size(size);
-  if (size_t(*buf_end - remote_buf.get()) > remote_buf.size()) {
-    FATAL() << "overflow";
+template <typename Arch = NativeArch>
```

Comment:
I prefer skipping the default parameter and just providing it explicitly where necessary. A little bit easier to read.